### PR TITLE
Collision object testing

### LINKED ISSUE #149: [FEATURE] CollisionObject Testing
### Description
Test methods of CollisionObject class.

### Estimated Effort
2

### PR DESCRIPTION
**Pull request by jwryan9**
## Description

Implementation of unit tests for CollisionObject class.

Closes #149 

## Type of change

- [x] Additional Testing

## How has this been tested?

All previous and new BoatOS tests pass.

## Code Coverage

Screen Shot 2019-04-22 at 10 08 43 PM

### CODE REVIEW

File: BoatOS/src/main/java/com/imarc/marc_i/arch/simulation/map/CollisionObject.java [hunk 1]
```diff
@@ -72,17 +72,22 @@ protected Point lineCircleCollision(LineCollision l, CircleCollision cir) {
     }
 
     protected Point lineLineCollision(LineCollision l1, LineCollision l2) {
-        double m1 = (l1.p2.y - l1.p1.y) / (l1.p2.x - l1.p1.x);
-        double m2 = (l2.p2.y - l2.p1.y) / (l2.p2.x - l2.p1.x);
+        double m1 = calculateSlope(l1);
```
Comment (cooperbell) [inline BoatOS/src/main/java/com/imarc/marc_i/arch/simulation/map/CollisionObject.java]: Call calculate scope up above for m1 on line 46

Comment (jwryan9) [reply BoatOS/src/main/java/com/imarc/marc_i/arch/simulation/map/CollisionObject.java]: Resolved with https://github.com/UIOWA5830SP19/SPP100/pull/150/commits/da4a6fae84cea340564d9ef602c829d224cea48f

**Comment by cooperbell** [comment]
Please test the [`collide()`](https://github.com/UIOWA5830SP19/SPP100/blob/3693dad57efc93ffa193fd1e7ebcacc47c82598d/BoatOS/src/main/java/com/imarc/marc_i/arch/simulation/map/CollisionObject.java#L19) method too

**Comment by jwryan9** [comment]
> Please test the [`collide()`](https://github.com/UIOWA5830SP19/SPP100/blob/3693dad57efc93ffa193fd1e7ebcacc47c82598d/BoatOS/src/main/java/com/imarc/marc_i/arch/simulation/map/CollisionObject.java#L19) method too

Fixed in https://github.com/UIOWA5830SP19/SPP100/pull/150/commits/9ae5e5b7212c9ef2856d32cb9d1e47f43395a091

**Comment by jwryan9** [comment]
> > Please test the [`collide()`](https://github.com/UIOWA5830SP19/SPP100/blob/3693dad57efc93ffa193fd1e7ebcacc47c82598d/BoatOS/src/main/java/com/imarc/marc_i/arch/simulation/map/CollisionObject.java#L19) method too
> 
> Fixed in [9ae5e5b](https://github.com/UIOWA5830SP19/SPP100/commit/9ae5e5b7212c9ef2856d32cb9d1e47f43395a091)

Coverage is now up to 31%

**Pull request merged**